FreeRTOS/FreeRTOS-Kernel
# Update SMP get idle task memory for static allocation

Add static idle task memory support for SMP

Description
-----------
In the FreeRTOS SMP, configNUMBER_OF_CORES - 1 idle tasks which do the minimal job are also created to ensure that each core has an idle task to run when no other task is available to run. The prototype of `vApplicationGetIdleTaskMemory()` is changed in SMP to specify the idle task memory.
```
void vApplicationGetIdleTaskMemory( StaticTask_t ** ppxIdleTaskTCBBuffer,
                                    StackType_t ** ppxIdleTaskStackBuffer,
                                    uint32_t * pulIdleTaskStackSize,
                                    BaseType_t xCoreID );
```

Set xCoreID to 0 to specify the idle task memory for the active idle task. Set xCoreID to 1 ~ ( configNUMBER_OF_CORES - 1 ) to specify the idle task memory for a passive idle task which does the minimal job.

codecov/project : fixed in https://github.com/FreeRTOS/FreeRTOS/pull/1096
FreeRTOS-Kernel Demos / GNU ARM Toolchain (pull_request) : fixed in https://github.com/FreeRTOS/FreeRTOS-Community-Supported-Demos/pull/13

Test Steps
-----------


Checklist:
----------



Related Issue
-----------
Address #750 comment in line 469


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
